### DISCUSSION TITLE
Improve Incremental adoption

### DISCUSSION THREAD

**Discussion by amesas** [vercel/next.js#30518] [Ideas]
### Describe the feature you'd like to request

Possibility to add headers to rewrites.

Not usable when you need to add Host headers, Authorization, API Keys, etc.

### Describe the solution you'd like

      {
        source: '/:path*',
        destination: `https://proxy.example.com/:path*`,
       headers: [......]
      },

### Describe alternatives you've considered

Maybe not only headers, but add params, etc...

Like has: [{ type: 'query', key: 'overrideMe' }]

But "add" instead of has

**Comment by gary-lev** [comment]
Yes this would be great for the cases where you want rewrite /api/something/* to an internal service, but that service requires an Authorization header to be added.  

The alternative seems to be a much clunkier catchall api route that proxies the api calls.
